packages/discovery-react-components/src/components/StructuredQuery/utils/getFieldNames.ts

```diff
@@ -0,0 +1,18 @@
+import DiscoveryV2 from 'ibm-watson/discovery/v2';
+import uniq from 'lodash/uniq';
+
+export const getFieldNames = (response: DiscoveryV2.ListFieldsResponse | null): string[] => {
+  if (response && response.fields) {
+    const fieldObjects = response.fields;
+    const fieldNames = fieldObjects.reduce((namesArray: string[], fieldObject) => {
+      if (fieldObject.type === 'nested') {
+        return namesArray;
+      } else if (fieldObject.field) {
+        namesArray.push(fieldObject.field);
+      }
+      return namesArray;
+    }, []);
+    return uniq(fieldNames);
```

Comment:
nit - you can just do `else if (fieldObject.field && !namesArray.includes(fieldObject.field)) {` instead of doing a uniq check

Reply:
Ah good call, thanks